### ISSUE TITLE
Js error when mouse over showing tooltip and c3 is destroyed

### ISSUE THREAD

**Issue by danelkhen** [masayuki0812/c3#997]
Sometimes I have to completely regenerate c3 graph (when update isn't supported well for some data changes), if the mouse is hovering over the graph during that time I get the following js error:
```
Uncaught TypeError: Cannot read property 'data_types' of null
```
The error can be tested using this code (just move the mouse and let the magic happen):
```
    var _chart;
    var _testIndex = 0;
    function test() {
        _testIndex++;
        console.info("testing "+_testIndex);
        if (_chart != null) {
            _chart.destroy();
            _chart = null;
        }
        var x = {
            data: { columns: [["x"], ["y"]], x: "x", }
        };
        for (var i = 1; i < 300; i++) {
            x.data.columns[0].push(i * 2);
            x.data.columns[1].push(i * 3);
        }
        _chart = c3.generate(x);
        if(_testIndex<20)
            scheduleTest();
    }

    test();
    function scheduleTest() {
        window.setTimeout(test, 1000);
    }
```
Happens in c3 0.4.9 (latest), and d3 3.5.0 (recommended version as far as I know)

Thanks

**Comment by morlay** [comment]
`.destroy()` make this issue, not sure why this happen, they just reset the internal, but when this issue happen, the new chart's internal have be change too.

     c3_chart_fn.destroy = function () {
        var $$ = this.internal;

        window.clearInterval($$.intervalForObserveInserted);
        window.onresize = null;

        $$.selectChart.classed('c3', false).html("");

        // MEMO: this is needed because the reference of some elements will not be released, then memory leak will happen.
        Object.keys($$).forEach(function (key) {
            $$[key] = null;
        });

        return null;
    };

**Comment by salemdar** [comment]
+1 for this issue. It doesn't effect anything as far as I can see, but it's unpleasant to see errors on the console.

**Comment by masayuki0812** [comment]
Thanks for reporting. Yeah, this should be fixed and I think now it's been fixed by the commit above.
I'll release this fix in the next release `0.4.10` shortly. Thanks.

**Comment by masayuki0812** [comment]
`0.4.10` has been released, so let me close at the moment. Thanks.
